### ISSUE TITLE
Invalid violation with the arrow-parens rule when using async on a function

### ISSUE THREAD

**Issue by glen-84** [palantir/tslint#1458]
### Bug Report

- __TSLint version__: `3.14.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `VS Code`

#### TypeScript code being linted

```ts
export const configure = async (aurelia: Aurelia) => {
    // ...
};
```

with `tslint.json` configuration:

```json
"arrow-parens": true
```

#### Actual behavior

`Parentheses are required around the parameters of an arrow function definition`

#### Expected behavior

No error/warning.

Related to #1442.

**Comment by JKillian** [comment]
cc @YuichiNukiyama. Good catch @glen-84, didn't think of this edge case!

**Comment by YuichiNukiyama** [comment]
I don't expect this edge case...
I'll try to fix it. Please wait for my PR.

**Comment by internetcowboy** [comment]
I see this issue still present in v3.15.1

**Comment by JKillian** [comment]
@internetcowboy I tested your sample code on my machine and can't reproduce what you're seeing. Can you double-check and make sure you're using `3.15.1` and the latest TypeScript?

**Comment by internetcowboy** [comment]
@JKillian strange. It's not the end of the world on my end but figured I would voice it. I still see the issue though after an uninstall/reinstall of tslint and typescript locally and globally. It could just be an issue with this box though.

```

xxx:server xxx$ npm list tslint
xxx-backend@1.0.0 /Users/xxx/Documents/development/xxx/server
└── tslint@3.15.1

xxx:server xxx$ npm list tslint -g
/usr/local/lib
└── tslint@3.15.1

xxx:server xxx$ npm list typescript -g
/usr/local/lib
└── typescript@2.0.2

xxx:server xxx$ npm list typescript
xxx-backend@1.0.0 /Users/xxx/Documents/development/xxx/server
├── typescript@2.0.2

```

**Comment by JKillian** [comment]
@internetcowboy double-checked on my end too, and still couldn't reproduce it. If you want, maybe post the whole function and the exact error message with line and column number?

Looks like the versions you have of both packages is correct.

**Comment by jeffijoe** [comment]
I'm getting this issue as well.

![image](https://cloud.githubusercontent.com/assets/2505474/20888246/564ff0e0-bafe-11e6-9248-a51548690c55.png)

**Comment by adidahiya** [comment]
@jeffijoe which versions of typescript and tslint are you using?

**Comment by jeffijoe** [comment]
@adidahiya

```
"typescript": "^2.2.0-dev.20161130"
"tslint": "^4.0.2"
```

**Comment by adidahiya** [comment]
I think @IllusionMH gets it right here -- https://github.com/palantir/tslint/issues/1572#issuecomment-249455843 -- this unfortunately a breaking change in the TS language services and we'll have to change the code to handle both cases.
